src/System.Private.CoreLib/shared/System/Text/UTF8Encoding.cs

```diff
@@ -715,12 +715,15 @@ internal override unsafe int GetByteCount(char* chars, int count, EncoderNLS bas
                     break;
 
                 LongCodeWithMask:
-#if BIGENDIAN
-                    // be careful about the sign extension
-                    ch = (int)(((uint)ch) >> 16);
-#else // BIGENDIAN
-                    ch = (char)ch;
-#endif // BIGENDIAN
+                    if (!BitConverter.IsLittleEndian)
```

Comment:
Nit: Could you please change flip conditions to `if (BitConverter.IsLittleEndian)` to make them easier to read and a bit less of work for JIT to optimize them out?